packages/components/src/internal/components/forms/BulkUpdateForm.test.tsx

```diff
@@ -0,0 +1,132 @@
+    name: 'fileInput',
+    caption: 'fileInput',
+    fieldKeyArray: ['fileInput'],
+    shownInUpdateView: true,
+    userEditable: true,
+    inputType: 'file',
+});
+const SCHEMA = 'samples';
+const QUERY = 'testST';
+const QUERY_INFO = QueryInfo.fromJsonForTests({
+    name: QUERY,
+    schemaName: SCHEMA,
+    columns: {
+        update: COLUMN_CAN_UPDATE,
+        neither: COLUMN_CANNOT_UPDATE,
+        fileInput: COLUMN_FILE_INPUT,
+    },
+});
+
+const DEFAULT_PROPS = {
+    onComplete: jest.fn,
+    onCancel: jest.fn,
+    onSubmitForEdit: jest.fn,
+    queryInfo: QUERY_INFO,
+    viewName: undefined,
+    selectedIds: [],
+    updateRows: (schemaQuery: SchemaQuery, rows: any[]) => Promise.resolve(),
+};
+
+const commonResults = {
+    "samples/testst": {
+        "127796": {
+            "update": {
+                "value": "abc"
+            },
+            "fileInput": {
+                "value": "/trunk/build/deploy/files/LKSM/@files/sampletype/test.txt",
+                "url": "/LKSM-dan/core-downloadFileLink.view?propertyId=82852",
+                "displayValue": "sampletype/test.txt"
+            }
+        },
+        "127797": {
+            "update": {
+                "value": "abc"
+            },
+            "fileInput": {
+                "value": "/trunk/build/deploy/files/LKSM/@files/sampletype/test.txt",
+                "url": "/LKSM-dan/core-downloadFileLink.view?propertyId=82852",
+                "displayValue": "sampletype/test.txt"
+            }
+        }
+    }
+};
+
+const commonResp = {
+    key: QUERY,
+    models: commonResults,
+    orderedModels: { 'samples/testst': fromJS(['127796', '127797']) },
+    queries: { [QUERY]: QueryInfo.fromJsonForTests({}) },
+    rowCount: 0,
+};
+
+const selectRowsDeprecated_ = selectRowsDeprecated as jest.Mock;
+
+jest.mock('../../query/api', () => ({
+    ...jest.requireActual('../../query/api'),
+    selectRowsDeprecated: () => createMockSelectRowsDeprecatedResponse(commonResp),
+}));
+
+
+describe('BulkUpdateForm', () => {
+    // TODO missing test cases for main functionality of component
+    describe('columnFilter', () => {
+
+        test('filters without uniqueKeyField', async () => {
+            let container;
+            await act(async () => {
+                container = render(<BulkUpdateForm {...DEFAULT_PROPS} />);
+            });
+
+            expect(document.querySelectorAll('.query-info-form')).toHaveLength(1);
+            expect(document.querySelectorAll('.toggle-group-icon')).toHaveLength(2)
+            expect(document.querySelectorAll('input#update')).toHaveLength(1);
+            expect(document.querySelector('input#update').getAttribute('value')).toBe('abc');
+            expect(document.querySelectorAll('.attachment-card__name')).toHaveLength(1);
+            expect(document.querySelector('.attachment-card__name').textContent).toBe('test.txt')
+        });
+
+        test('filters with uniqueFieldKey', async () => {
```

Comment:
There are still ajax requests being made that are subsequently failing which can cause flakiness in tests. I've narrowed down where the requests are coming from and it's in `QueryInfoForm` requesting audit settings. 

I think the minimal change here would be to optionally accept an `api?: ComponentsAPIWrapper` on `BulkUpdateForm` and pass through to the `QueryInfoForm` so that it can be mocked out in tests.

Another, and more ideal, solution is to create a functional wrapper in `QueryInfoForm` that gets the API from context rather than using the defaultProps approach. Then `renderWithAppContext()` could be utilized and everything would be mocked appropriately.